src/main/java/twilightforest/client/renderer/TileEntityTFChestRenderer.java

```diff
@@ -0,0 +1,155 @@
+package twilightforest.client.renderer;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockChest;
+import net.minecraft.client.model.ModelChest;
+import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.util.ResourceLocation;
+
+import org.lwjgl.opengl.GL11;
+import org.lwjgl.opengl.GL12;
+
+import cpw.mods.fml.common.FMLLog;
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+import twilightforest.TwilightForestMod;
+import twilightforest.block.BlockTFChest;
+import twilightforest.block.BlockTFChest.WoodType;
+import twilightforest.client.model.ModelTFChest;
+import twilightforest.client.model.ModelTFLargeChest;
+import twilightforest.tileentity.TileEntityTFChest;
+
+@SideOnly(Side.CLIENT)
+public class TileEntityTFChestRenderer extends TileEntitySpecialRenderer {
+
+    public static final String[] names = { "twilight", "canopy", "mangrove", "darkwood", "time", "trans", "mining",
+            "sort" };
+    private static final ResourceLocation[] doubleTextures = new ResourceLocation[names.length];
+    private static final ResourceLocation[] textures = new ResourceLocation[names.length];
+    private ModelChest modelChest = new ModelTFChest();
+    private ModelChest modelLargeChest = new ModelTFLargeChest();
+
+    public TileEntityTFChestRenderer() {
+        for (int i = 0; i < names.length; i++) {
+            doubleTextures[i] = new ResourceLocation(TwilightForestMod.MODEL_DIR + "chest/" + names[i] + "_double.png");
+            textures[i] = new ResourceLocation(TwilightForestMod.MODEL_DIR + "chest/" + names[i] + ".png");
+        }
+    }
+
+    public void renderTileEntityAt(TileEntityChest tileEntity, double x, double y, double z, float f) {
+        int i;
+
+        if (!tileEntity.hasWorldObj()) {
+            i = 0;
+        } else {
+            Block block = tileEntity.getBlockType();
+            i = tileEntity.getBlockMetadata();
+
+            if (block instanceof BlockChest && i == 0) {
+                try {
+                    ((BlockChest) block).func_149954_e(
+                            tileEntity.getWorldObj(),
+                            tileEntity.xCoord,
+                            tileEntity.yCoord,
+                            tileEntity.zCoord);
+                } catch (ClassCastException e) {
+                    FMLLog.severe(
+                            "Attempted to render a chest at %d,  %d, %d that was not a chest",
+                            tileEntity.xCoord,
+                            tileEntity.yCoord,
+                            tileEntity.zCoord);
+                }
+                i = tileEntity.getBlockMetadata();
+            }
+
+            tileEntity.checkForAdjacentChests();
+        }
+
+        BlockTFChest.WoodType type = (((TileEntityTFChest) tileEntity).cachedMaterial == WoodType.NULL)
```

Comment:
This is not thread safe and is throwing class cast exceptions whenever you break a chest with angelica installed.
Double chests also render incorrectly when you break one of the sides and will either keep rendering both chests or render none of them.

Reply:
Oh, yes, this thing
Totally forgot about it